### PR TITLE
Feature/global transform and optimizations

### PR DESCRIPTION
**Pull request by GlynLeine**

### CODE REVIEW

File: legion/engine/core/ecs/entity_handle.hpp [hunk 1]
```diff
@@ -100,20 +103,25 @@ namespace legion::core::ecs
...
+            return valid();
+        }
 
         /**@brief Returns the id of the entity this handle references.
          * @returns id_type If the handle is valid it will return the entity id, otherwise invalid_id.
          */
         L_NODISCARD id_type get_id() const;
 
+        L_NODISCARD entity_set children() const;
+
```
Comment (Algo-ryth-mix) [inline legion/engine/core/ecs/entity_handle.hpp]: While children is quite self-explanatory, it still could use a doxygen comment

File: legion/engine/rendering/pipeline/default/stages/meshbatchingstage.cpp [hunk 2]
```diff
@@ -88,86 +89,110 @@ namespace  legion::rendering
...
+        //        }
+
+        //        material = renderable.get_material();
+        //        model = renderable.get_model();
+
+        //        m_instances[entity] = std::make_pair(material, model);
+        //        (*batches)[material][model].insert(entity);
+        //    }
+        //    m_toReinsert.clear();
+        //}
```
Comment (Algo-ryth-mix) [inline legion/engine/rendering/pipeline/default/stages/meshbatchingstage.cpp]: this file could use a little cleanup

**Review by Algo-ryth-mix — CHANGES_REQUESTED** [review]

File: legion/engine/core/defaults/hierarchysystem.cpp [hunk 3]
```diff
@@ -0,0 +1,44 @@
+#include <core/defaults/hierarchysystem.hpp>
+
+void legion::core::HierarchySystem::onPositionModified(events::component_modification<position>* event)
+{
+    /*if (event->entity == m_ignore)
+        return;*/
+
```
Comment (Jelled1st) [inline legion/engine/core/defaults/hierarchysystem.cpp]: This could probably go

File: legion/engine/core/defaults/hierarchysystem.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+void legion::core::HierarchySystem::onRotationModified(events::component_modification<rotation>* event)
+{
+    rotation diff = event->newValue * math::inverse(event->oldValue);
+    position p = event->entity.read_component<position>();
+    for (auto child : event->entity.children())
+    {
+        //m_ignore = child;
+        child.write_component<position>(p + (math::toMat3(diff) * (child.read_component<position>() - p)));
+        //m_ignore = invalid_id;
```
Comment (Jelled1st) [inline legion/engine/core/defaults/hierarchysystem.cpp]: These comments can probably go. More comments up ahead in this file. I will not commend on these

**Review by Developer-The-Great — APPROVED** [review]

**Review by Jelled1st — APPROVED** [review]
There was some uncommented code, but since you said you removed most of it during my review, I'll accept it.

**Review by Algo-ryth-mix — APPROVED** [review]

**Pull request merged**